Services/UI/OJS.Services.Ui.Data/Implementations/SubmissionsDataService.cs

```diff
@@ -26,107 +27,107 @@ public Task<IEnumerable<TServiceModel>> GetLatestSubmissions<TServiceModel>(int
             .ToEnumerableAsync();
 
     public Submission? GetBestForParticipantByProblem(int participantId, int problemId) =>
-            this.GetAllByProblemAndParticipant(problemId, participantId)
-                .Where(s => s.Processed)
-                .OrderByDescending(s => s.Points)
-                .ThenByDescending(s => s.Id)
-                .FirstOrDefault();
+        this.GetAllByProblemAndParticipant(problemId, participantId)
+            .Where(s => s.Processed)
+            .OrderByDescending(s => s.Points)
+            .ThenByDescending(s => s.Id)
+            .FirstOrDefault();
+
+    public IQueryable<Submission> GetByIdQuery(int id) =>
+        this.DbSet
+            .Where(s => s.Id == id);
+
+    public IQueryable<Submission> GetAllByProblem(int problemId)
+        => base.DbSet.Where(s => s.ProblemId == problemId);
+
+    public IQueryable<Submission> GetAllByProblemAndParticipant(int problemId, int participantId) =>
+        this.GetQuery(
+            filter: s => s.ParticipantId == participantId && s.ProblemId == problemId,
+            orderBy: q => q.CreatedOn,
+            descending: true);
+
+    public async Task<IEnumerable<Submission>> GetAllByProblemAndUser(int problemId, string userId, int? take = Constants.Submissions.DefaultCount)
+        => await this.GetQuery(
+                filter: s => s.ProblemId == problemId && s.Participant.UserId == userId,
+                take: take)
+            .ToEnumerableAsync();
 
-        public IQueryable<Submission> GetByIdQuery(int id) =>
+    public IQueryable<Submission> GetAllFromContestsByLecturer(string lecturerId) =>
+        this.DbSet
+            .Include(s => s.Problem!.ProblemGroup.Contest.LecturersInContests)
+            .Include(s => s.Problem!.ProblemGroup!.Contest!.Category!.LecturersInContestCategories)
```

Comment:
We do not use includes. This is handled by automapper

Reply:
Removed.

Reply:
This is also an old method.